### ISSUE TITLE
Grid Colors on Linux

### ISSUE THREAD

**Issue by toxpal** [facelessuser/Rummage#254]
I experience a major issue with grid in Files field. To summarize, every other line in grid is white text on white background. Screenshot - https://s33.postimg.cc/ds48rdidb/rummage_dark.png

First, I thought it might be some bug in my color scheme because I use KDE Dark theme (even if other apps with grid files layout, like Beyond Compare, work well). So I changed theme to light, but issue still remains - https://postimg.cc/image/p4gu9er4r/ 

As you see in 2nd screen, program window is still dark (but color of titlebar is now different) and problem remains.

In short, no matter which color scheme I choose in OS settings, Rummage still uses the old one (KDE Dark, which was used during Rummage installation) for background color. Other apps switch to new color scheme immediately (even without closing apps), but I can't change Rummage colors even after multiple restarts.

I realize that might need lots of debugging, so maybe there's a possibility to have an option for disabling grid layout (with different lines)? Even if that doesn't change the background color of Rummage, it would solve the issue.

**Comment by facelessuser** [comment]
Hmm. I could force the text and all column backgrounds to be a safe color. I could disable alternating colors... Maybe I could detect cases when the font color is light, and force it to properly contrast. I'll have to see if I can read such info via WxPython. I'm glad that that seems to be the only issue with dark themes. I think listctrls is the only place I force specific colors.

Thanks for the report, this is definitely something that should get fixed. You're honestly probably the first to use it with a dark theme :smile:.

**Comment by facelessuser** [comment]
For clarifications, rummage uses wxpyhon default colors that are accepted from the system. I don't manually set any specific colors. I guess I could control this and force themes for people who want a light or dark theme. Forget down the road I could even enable custom themes.

I don't know. First step is to most likely provide a stop gap solution.

**Comment by facelessuser** [comment]
Maybe, I'm not sure if I even have a way to force a refresh of color cache. If the system is rebooted with a new theme color, does rummage finally accept it?

Most likely I'll have to hunt around and see if other developers have hit this issue work WxPython.

For now I'll "attempt" to read default background and see if I can synthesize an alternate color that works. If that doesn't work, I can most likely just allow disabling alternating colors.

**Comment by facelessuser** [comment]
As this creates a usability issue, I'll probably fast track the next release. I wanted to address the en_US issue, but I'm having a difficult time pin pointing where that originates, so that might have to wait.

**Comment by toxpal** [comment]
Unfortunately, no. Reboots don't help, for some reason background color (also other colors like text color, selected item color, etc.) of Rummage always remains the same which was used during Rummage's installation. However, titlebar color is changed immediately right after I activate new scheme in OS settings.

**Comment by facelessuser** [comment]
WxPython doesn't really let you control title bar, so that doesn't surprise me. It's probably silly controlled by the os.

I'll look into this and see if there is a way to get WxPython to respect theme colors, but for now, I'm probably just going to directly after the listctrls.

Ahh, the fun of cross platform support...

**Comment by facelessuser** [comment]
Okay, looking at this, it appears there is built in support which *should* pick the appropriate color for the zebra stripe on a given system via https://wxpython.org/Phoenix/docs/html/wx.ListCtrl.html#wx.ListCtrl.EnableAlternateRowColours.  Apparently I must have missed this, but this provides an easy solution.

**Comment by facelessuser** [comment]
This won't change the fact that Rummage doesn't respect different themes, but it should give sane alternating rows.

**Comment by facelessuser** [comment]
This should be in the latest 4.2.0 release.  Hopefully that is all the breaking things.
